src/linear_algebra/vectors.lean

```diff
@@ -0,0 +1,56 @@
+/-
```

Comment:
I think the lemmas in this file just belong in `data.matrix.notation` with the other lemma about dot_product.

Reply:
@kmill 
What do you think?

Reply:
I mean @kmill recommended me to put the lemmas into the new file. So I want to wait for his opinion before I move it somewhere else.

Reply:
Thanks for asking. I'm fine with them going anywhere appropriate. My reasoning was that `linear_algebra/vectors` could be a repository for things that are useful when you want to manipulate concrete vectors in low dimensions, but if @eric-wieser thinks it's still better for them to go into `data/matrix/notation`, I'll defer to that opinion.

Reply:
Actually, there might be a problem having it in `data/matrix/notation` since it uses the dot for matrix multiplication instead of dot product.

Two solutions are to either (1) leave things as set out by this PR or (2) remove the dot product notation and replace it with the `dot_product` function.

Reply:
I prefer (1). The dot product is used many times and calling it as a function is much less readable. Do we want to wait for another comment or should I mark this as resolved?

Reply:
I don't expect there to be an issue with the notation if you put it in the other file.

Reply:
Yeah, it is what we agreed on, I suppose. It will stay in `src/linear_algebra/vectors.lean`.

Reply:
What Eric said is ambiguous here :smile:  He thinks I'm wrong about there being any problems having it in `data/matrix/notation`.

I can check later if you want. (I have some teaching duties to do first.)

Reply:
> He thinks I'm wrong about there being any problems having it in data/matrix/notation.

Yes; or at least, I think any such problems can be easily resolved by adding a `section` around the separate matrix and vector locales.

I'm optimistic though that lean's rarely-used operator overloading might just do the right thing.

Reply:
Let's wait for #11241 to be merged before refactoring this.

Reply:
> Actually, there might be a problem having it in `data/matrix/notation` since it uses the dot for matrix multiplication instead of dot product.

> I'm optimistic though that lean's rarely-used operator overloading might just do the right thing.

The collision was a problem. Hence I introduced `•` for `dot_product` (that is `\bullet` whereäs `\cdot` stays for matrix multiplication).

Reply:
Changed to ` ⬝ᵥ `.